crates/oxc_linter/src/rules/oxc/only_used_in_recursion.rs

```diff
@@ -60,24 +60,40 @@ declare_oxc_lint!(
 
 impl Rule for OnlyUsedInRecursion {
     fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
-        let AstKind::Function(function) = node.kind() else {
-            return;
-        };
-
-        let Some(function_id) = &function.id else {
-            return;
+        let (function_id, function_parameters) = match node.kind() {
+            AstKind::Function(function) => {
+                if function.is_typescript_syntax() {
+                    return;
+                }
+                if let Some(function_id) = &function.id {
+                    (function_id, &function.params)
+                } else {
+                    return;
+                }
+            }
+            AstKind::ArrowFunctionExpression(arrow_function) => {
+                let Some(parent) = outermost_paren_parent(node, ctx) else {
+                    return;
+                };
+                if let AstKind::VariableDeclarator(decl) = parent.kind() {
+                    if let Some(ident) = decl.id.get_binding_identifier() {
+                        (ident, &arrow_function.params)
+                    } else {
```

Comment:
remove `else returns`, just have a `return` after the first `if`. applies to the `match` branch above as well.